src/main/java/hudson/plugins/s3/S3CopyArtifact.java

```diff
@@ -217,7 +222,7 @@ public void perform(@Nonnull Run<?, ?> dst, @Nonnull FilePath targetDir, @Nonnul
 
             excludeFilter = env.expand(excludeFilter);
 
-            if (src instanceof MavenModuleSetBuild) {
+            if (isMavenPluginInstalled() && src instanceof MavenModuleSetBuild) {
```

Comment:
If the plugin is not installed, can the `src` be an instance of `MavenModelSetBuild`? This seems redundant, no?